client.go

```diff
@@ -8,36 +8,34 @@ import (
 	"net/url"
 
 	"github.com/google/go-querystring/query"
-	"github.com/warrant-dev/warrant-go/v4/client"
-	"github.com/warrant-dev/warrant-go/v4/config"
 )
 
 type Client struct {
-	warrantClient *client.WarrantClient
+	warrantClient *WarrantClient
 }
 
-func NewClient(config config.ClientConfig) Client {
+func NewClient(config ClientConfig) Client {
```

Comment:
Let's add the ability for developers to override the default http client if they want:
```go
func NewClient(config ClientConfig) WarrantClient {
    return WarrantClient{
        apiClient: NewApiClient(config),
    }
}
```